execution_engine/src/storage/global_state/lmdb.rs

```diff
@@ -281,16 +288,43 @@ impl StateProvider for LmdbGlobalState {
         correlation_id: CorrelationId,
         trie_keys: Vec<Digest>,
     ) -> Result<Vec<Digest>, Self::Error> {
-        let txn = self.environment.create_read_txn()?;
-        let missing_descendants =
-            missing_trie_keys::<Key, StoredValue, lmdb::RoTransaction, LmdbTrieStore, Self::Error>(
+        if trie_keys
+            .iter()
+            .filter(|digest| {
+                !self
+                    .digests_without_missing_descendants
+                    .read()
+                    .unwrap()
+                    .contains(digest)
```

Comment:
Actually, isn't this going to slow this check down substantially, if we are locking for every single digest? I think you should hoist the read() call to above and then call contains.

Reply:
Good catch!
Fixed in aac1d6b4d9dba612378a1af8429fe70f09415908

Reply:
I've looked at the call sites and we always call this method (`missing_trie_keys`) with just one `Digest`. This means that it's either zero or single trie key we will call the `missing_trie_keys` with.